ggshield/core/git_hooks/ci/previous_commit.py

```diff
@@ -282,6 +303,25 @@ def azure_pull_request_previous_commit_sha(verbose: bool) -> Optional[str]:
     return None
 
 
+def get_new_branch_before_sha(branch_name: str, verbose: bool) -> Optional[str]:
```

Comment:
This name is not very clear to me. Maybe get_new_branch_parent_commit()?

Also: can you add a docstring?